internal/cloud/backend.go

```diff
@@ -413,7 +414,15 @@ func (b *Cloud) discover() (*url.URL, error) {
 
 	host, err := b.services.Discover(hostname)
 	if err != nil {
-		return nil, err
+		var serviceDiscoErr *disco.ErrServiceDiscoveryNetworkRequest
+
+		switch {
+		case errors.As(err, &serviceDiscoErr):
+			err = fmt.Errorf("a network issue prevented cloud configuration; %w", err)
+			return nil, err
+		default:
+			return nil, err
+		}
```

Comment:
I see a couple of code issues (no need for locals, semicolon typo). Should we add the "cloud configuration" context to the other types of errors? Or do we want to lead with "service discovery" errors in that case?

```suggestion
		if errors.Is(err, disco.ErrServiceDiscoveryNetworkRequest) {
			return nil, fmt.Errorf("a network issue prevented cloud configuration: %w", err)
		default:
			return nil, fmt.Errorf("cloud configuration failed: %w", err)
		}
```